src/app/components/SingleCompositionTokenForm.tsx

```diff
@@ -1,99 +1,108 @@
-      let values = tokens;
-      const newToken = { ...tokens[index], property };
-      values.splice(index, 1, newToken);
-      setValue(values);
-    } else {
-      setValue({ ...tokens, property });
-    }
-    setMenuOpened(false);
-  }, [tokens]);
+  const seed = useUIDSeed();
+
+  const onPropertySelected = useCallback((newProperty: string) => {
+    // keep the order of the properties when select new property
+    const newOrderObj: NodeTokenRefMap = {};
+    const keysInTokenValue = Object.keys(tokenValue);
+    keysInTokenValue.splice(index, 1, newProperty);
+    keysInTokenValue.forEach((key, index) => {
+      newOrderObj[key as keyof typeof Properties] = String(index);
+    });
+    setOrderObj(newOrderObj);
 
+    // set newTokenValue
+    delete tokenValue[property as keyof typeof Properties];
+    tokenValue[newProperty as keyof typeof Properties] = value;
+    setTokenValue(tokenValue);
+    setError(null);
+  }, [tokenValue]);
 
   const onAliasChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-    if (Array.isArray(tokens)) {
-      let values = tokens;
-      const newToken = { ...tokens[index], value: e.target.value };
-      values.splice(index, 1, newToken);
-      setValue(values);
-    } else {
-      setValue({ ...tokens, value: e.target.value });
-    }
-  }, [tokens]);
+    tokenValue[property as CompositionTokenProperty] = e.target.value;
+    setTokenValue(tokenValue);
+  }, [tokenValue]);
 
   const handleToggleMenu = useCallback(() => {
     setMenuOpened(!menuOpened);
   }, [menuOpened]);
 
   const handleRemove = useCallback(() => {
-    onRemove(index);
-  }, [onRemove, index]);
-  
+    onRemove(property);
+  }, [onRemove, property]);
+
   return (
     <Box>
       <Box css={{
         display: 'flex',
         justifyContent: 'space-between',
         alignItems: 'center',
+        gap: '$3',
         '& > label': {
-          flex: 2,
+          flex: 4,
           fontSize: '$5 !important',
           '& > div > input': {
-            flex: 2,
-            marginRight: '$5',
-            height: '$10'
-          }
-        }
+            height: '$10',
+          },
+        },
       }}
       >
-        <PropertySwitchMenu open={menuOpened} onOpenChange={handleToggleMenu}>
-          <PropertySwitchMenuMainTrigger>
-            <span>{token.property}</span>
-          </PropertySwitchMenuMainTrigger>
-          <PropertySwitchMenuContent sideOffset={2}>
-            <PropertySwitchMenuRadioGroup value={token.property}>
+        <DropdownMenu open={menuOpened} onOpenChange={handleToggleMenu}>
+          <DropdownMenuTrigger css={{ flex: 3, minHeight: '$10', border: `1px solid ${error && property.length === 0 ? '$fgDanger' : '$borderMuted'}` }}>
+            <span>{property}</span>
+          </DropdownMenuTrigger>
+          <DropdownMenuContent sideOffset={2} className="content scroll-container" css={{ maxHeight: '$11', backgroundColor: '$bgDefault', color: '$text' }}>
```

Comment:
```suggestion
          <DropdownMenuContent sideOffset={2} className="content scroll-container" css={{ maxHeight: '$dropdownMaxHeight' }}>
```

No need to specify bgColor, the dropdown handles this for us. Also I changed name of token to be descriptive.